src/server/cluster/cluster_family.cc

```diff
@@ -227,13 +238,11 @@ void ClusterShardsImpl(const ClusterShardInfos& config, SinkReplyBuilder* builde
 }  // namespace
 
 void ClusterFamily::ClusterShards(SinkReplyBuilder* builder, ConnectionContext* cntx) {
-  if (IsClusterEmulated()) {
-    return ClusterShardsImpl({GetEmulatedShardInfo(cntx)}, builder);
-  } else if (tl_cluster_config != nullptr) {
-    return ClusterShardsImpl(GetConfigForStats(cntx), builder);
-  } else {
-    return builder->SendError(kClusterNotConfigured);
+  auto shard_infos = GetShardInfos(cntx);
+  if (shard_infos) {
```

Comment:
Super nit, but I think it'd be slightly more elegant to first return on errors (i.e. `if (!shard_infos) { SendError(); }`)

Reply:
i agree, i just automatically followed the previous flow.